Bluetooth: Mesh: Fix unref null pointer

LingaoM opened the pull request:
When pb-gatt advertising enabled, after establish connection,
will call `cb->connected` and `cb->adv_send`.

In previous connected also clear `ADV_FLAG_PROXY` flag, but
in `adv_send` will attempt unref null point buffers.

Signed-off-by: Lingao Meng <menglingao@xiaomi.com>

LingaoM committed:
Bluetooth: Mesh: Fix unref null pointer

When pb-gatt advertising enabled, after extablish connect,
will call `cb->connected` and `cb->adv_send`.

In previous connected also clear `ADV_FLAG_PROXY` flag, but
in `adv_send` will attempt unref null point buffers.

Signed-off-by: Lingao Meng <menglingao@xiaomi.com>

trond-snekvik commented on subsys/bluetooth/mesh/adv_ext.c:
```diff
@@ -343,8 +343,9 @@ static void adv_sent(struct bt_le_ext_adv *instance,
 
-	if (!atomic_test_and_clear_bit(adv->flags, ADV_FLAG_PROXY)) {
+	if (adv->buf) {
 		net_buf_unref(adv->buf);
```
Perhaps we could clear adv->buf here now too, to ensure there are no regressions for this check in the future?

LingaoM replied:
Logically, calling here is already an error

LingaoM committed:
Bluetooth: Mesh: Fix adv_send call before enabled.

Since the default syswork thread priority = `-1`.
`adv_send` will call when controller report advertising
sending completed, due to this process by BT RX task, will
maybe process this before `buf_send`, since, sysworkq will be
used by other place and defer by any place.

Note: secure beacon will be 40ms, and friend and lpn will be 20ms.

This problem is very easy to reproduce, especially in native posix.

Signed-off-by: Lingao Meng <menglingao@xiaomi.com>

LingaoM commented on subsys/bluetooth/mesh/adv_ext.c:
```diff
@@ -208,6 +210,29 @@ static void send_pending_adv(struct k_work *work)
 
 	adv = CONTAINER_OF(work, struct bt_mesh_ext_adv, work.work);
+
+	if (atomic_test_and_clear_bit(adv->flags, ADV_FLAG_SENDED)) {
```
@trond-snekvik Please note that this is a new bug fix that I put in the same PR as it is associated with the first commit.

alxelax commented on subsys/bluetooth/mesh/adv_ext.c:
```diff
@@ -33,6 +33,8 @@
 	/** Controller is currently advertising */
 	ADV_FLAG_ACTIVE,
+	/** Advertising sending completed */
+	ADV_FLAG_SENDED,
```
```suggestion
	ADV_FLAG_SENT,
```
